extensions/data-plane/data-plane-azure-storage/src/test/java/org/eclipse/edc/connector/dataplane/azure/storage/pipeline/AzureStorageDataSinkTest.java

```diff
@@ -204,7 +204,47 @@ void getDestinationBlobName_shouldBeProperlyConcatenated(String blobName, String
                 .blobName(altName)
                 .request(request.build())
                 .build();
-        assertEquals(expected, sink.getDestinationBlobName(blobName));
+        assertEquals(expected, sink.getDestinationBlobName(blobName, 1));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = { "partName, blobName, folderName, 1, folderName/blobName",
+            "partName, blobName, folderName, 2, folderName/partName",
+            "partName, null, folderName, 1, folderName/partName",
+            "partName, null, folderName, 2, folderName/partName",
+            "partName, ''  , folderName, 1,  folderName/partName",
+            "partName, ''  , folderName, 2,  folderName/partName",
+            "partName, blobName,  folderName/, 1,  folderName/blobName",
+            "partName, blobName,  folderName/, 2,  folderName/partName",
+            "partName, null, folderName/, 1, folderName/partName",
+            "partName, null, folderName/, 2, folderName/partName",
+            "partName, ''  , folderName/, 1, folderName/partName",
+            "partName, ''  , folderName/, 2, folderName/partName",
+            "partName, blobName,  null, 1, blobName",
+            "partName, blobName,  null, 2, partName",
+            "partName, blobName,  '', 1, blobName",
+            "partName, blobName,  '', 2, partName",
+            "partName, '',   '', 1, partName" ,
+            "partName, '',   '', 2, partName" }
+            ,nullValues = { "null" })
+    void getDestinationBlobName_shouldDifferIfSingleOrMultipleFiles(String partName, String blobName, String folderName, int partSize, String expected) {
```

Comment:
this test is too hard to read and the scope too broad, please split it for various cases based on the condition tested, e.g.:
- `shouldReturnBlobName_whenFolderNameIsNullOrEmpty`
- `shouldAppendSlashToFolderName_whenItIsMissing`
- ...

looking at the tests there's some smell here: making a method "package-private" only for testing doesn't look good, likely that behavior (define destination name) would stay better in a separate component that's injected in the `DataSink` so it can be tested separately.
Please extract it and clean up test a bit.

Reply:
Regarding extracting the logic for the destination name. I've created a new class, but I don't know if this is the most appropriate approach. Please let me know if this is exactly what you want.